tests/test_enterprise/api/test_views.py

```diff
@@ -2696,6 +2700,226 @@ def test_bulk_enrollment(self, body, expected_code, expected_body):
+                        'license_uuid': '5a88bdcade7c4ecb838f8111b68e18ac'
+                    },
+                    {
+                        'email': 'xyz@test.com',
+                        'course_run_key': 'course-v1:edX+DemoX+Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '2c58acdade7c4ede838f7111b42e18ac'
+                    },
+                    {
+                        'email': 'abc@test.com',
+                        'course_run_key': 'course-v2:edX+DemoX+Second_Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '5a88bdcade7c4ecb838f8111b68e18ac'
+                    },
+                    {
+                        'email': 'xyz@test.com',
+                        'course_run_key': 'course-v2:edX+DemoX+Second_Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '2c58acdade7c4ede838f7111b42e18ac'
+                    },
+                ]
+            },
+            'expected_code': 202,
+            'expected_response': {
+                'successes': [],
+                'pending': [
+                    {'email': 'abc@test.com', 'course_run_key': 'course-v1:edX+DemoX+Demo_Course'},
+                    {'email': 'xyz@test.com', 'course_run_key': 'course-v1:edX+DemoX+Demo_Course'},
+                    {'email': 'abc@test.com', 'course_run_key': 'course-v2:edX+DemoX+Second_Demo_Course'},
+                    {'email': 'xyz@test.com', 'course_run_key': 'course-v2:edX+DemoX+Second_Demo_Course'}
+                ],
+                'failures': []
+            },
+            'expected_num_pending_licenses': 4,
+        },
+    )
+    @ddt.unpack
+    @mock.patch('enterprise.api.v1.views.EnterpriseCustomerViewSet._create_ecom_orders_for_enrollments')
+    # pylint: disable=unused-argument
+    def test_bulk_enrollment_in_bulk_courses(
+        self,
+        mock_ecom_order,
+        body,
+        expected_code,
+        expected_response,
+        expected_num_pending_licenses
+    ):
+        """
+        Tests the bulk enrollment endpoint at enroll_learners_in_courses.
+        """
+        factories.EnterpriseCustomerFactory(
+            uuid=FAKE_UUIDS[0],
+            name="test_enterprise"
+        )
+
+        permission = Permission.objects.get(name='Can add Enterprise Customer')
+        self.user.user_permissions.add(permission)
+
+        response = self.client.post(
+            settings.TEST_SERVER + ENTERPRISE_CUSTOMER_BULK_ENROLL_LEARNERS_IN_COURSES_ENDPOINT,
+            data=json.dumps(body),
+            content_type='application/json',
+        )
+        self.assertEqual(response.status_code, expected_code)
+        if expected_response:
+            response_json = response.json()
+            self.assertEqual(expected_response, response_json)
+        self.assertEqual(len(PendingEnrollment.objects.all()), expected_num_pending_licenses)
```

Comment:
just to be even stricter, always better to check this length before performing the operation too.. 0 is what we expect in this case I guess?